src/build/arm_quickstart/mainTemplate.json

```
@@ -0,0 +1,123 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "service_principal_id": {
```

```suggestion
      "servicePrincipalId": {
```

For this variable, and the others in this template, [ARM templates prefer camelCasing](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#general-recommendations-for-parameters) and will generate space separated values in the UI and other tools when interpreting them, e.g. `servicePrincipalId` will render `Service Principal ID`